### ISSUE TITLE
handyman clean never calls me back

### ISSUE THREAD

**Issue by joe-cygnarowicz-tr** [kenzanlabs/pipeline-handyman#26]
## Issue:

The documentation for clean shows:
`handyman.clean(['.dest/'], doneCallbackFunction);`

Clean is coded as:

```
function clean(path, done) {
  return del.sync(path, done);
}
```

https://www.npmjs.com/package/del#delsyncpatterns-options shows the signature for del.sync() to be
`del.sync(patterns, [options])`

=> thus, the "handyman clean done callback" is actually an options object for `del.sync()`.
## Potential Solutions:

1) This issue might be addressed by changing `clean()` as follows (and updating the signature in the documentation depending on whether options are supported or not):

```
function clean(path, options, done) {
  del.sync(path, options);
  done();
}
```

2) Another approach might be to announce *through its name) that `clean()` is synchronous (and update the signature in the documentation depending on whether options are supported or not):

```
function cleanSync(path, options) {
  // this actually "Returns an array of deleted paths."
  return del.sync(path, options);
}
```

It would then be up to developers to account for a callback within their gulp task.

3) Note that the test also needs to be updated. It currently calls
`handyman.clean([dir], null, true);`

**Comment by franciov** [comment]
I confirm this is an issue, and would consider a solution based on [Promises](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Promise) as well:
```
var handyman = require('pipeline-handyman');

handyman
  .clean(['.dest/'])
  .then((path) => {
    // success
  })
  .catch((err) => {
    // something wrong happened
  });
```

Note that `del` supports Promises: https://www.npmjs.com/package/del#usage

**Comment by obuckley-kenzan** [comment]
seems like there is a chance to kill two birds with one stone here, #23

**Comment by thescientist13** [comment]
@paul-barry-kenzan does this get resolved now via your work in #23 / #29 ?

**Comment by paul-barry-kenzan** [comment]
@thescientist13 @obuckley-kenzan 

Yes, it does. 

@joe-cygnarowicz-tr , thoughts on this since you were the OP?
